marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/4/triggers/entity-model-trigger.xqy

```diff
@@ -132,8 +132,15 @@ return (
       }, map:entry("database", xdmp:schema-database())
     ),
 
+  (: Check if TDE generation is disabled. Any value other than false or "false" disables TDE generation. :)
+  let $primary-type-def := $definitions => map:get($entity-title)
+  let $primary-type-def := if (fn:exists($primary-type-def)) then $primary-type-def
```

Comment:
An XQuery trick that I think works here is:

    let $primary-type-def := ($definitions => map:get($entity-title), $definitions => map:get(map:keys($definitions)[1]))[1]

i.e. return the first non-empty thing